keras_nlp/models/bert.py

```diff
@@ -0,0 +1,287 @@
+        x = keras.layers.Add(
+            name="embeddings/sum",
+        )((token_embedding, position_embedding, segment_embedding))
+        x = keras.layers.LayerNormalization(
+            name="embeddings/layer_norm",
+            axis=-1,
+            epsilon=1e-12,
+            dtype=tf.float32,
+        )(x)
+        x = keras.layers.Dropout(
+            dropout,
+            name="embeddings/dropout",
+        )(x)
+
+        # Apply successive transformer encoder blocks.
+        for i in range(num_layers):
+            x = TransformerEncoder(
+                num_heads=num_heads,
+                intermediate_dim=intermediate_dim,
+                activation=lambda x: keras.activations.gelu(
+                    x, approximate=True
+                ),
+                dropout=dropout,
+                kernel_initializer=initializer_fn,
+                name="transformer/layer_%d" % i,
+            )(x, padding_mask=input_mask)
+
+        # Construct the two BERT outputs. The pooled output is a dense layer on
+        # top of the [CLS] token.
+        sequence_output = x
+        pooled_output = keras.layers.Dense(
+            hidden_size,
+            activation="tanh",
+            name="pooled_dense",
+        )(x[:, CLS_INDEX, :])
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs={
+                "input_ids": token_id_input,
+                "segment_ids": segment_id_input,
+                "input_mask": input_mask,
+            },
+            outputs={
+                "sequence_output": sequence_output,
+                "pooled_output": pooled_output,
+            },
+            **kwargs,
+        )
+        # All references to `self` below this line
+        self.token_embedding = token_embedding_layer
+        self.initializer_fn = initializer_fn
+        self.vocab_size = vocab_size
+        self.hidden_size = hidden_size
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.max_sequence_length = max_sequence_length
+        self.num_segments = num_segments
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
+
+    def get_embedding_table(self):
+        return self.token_embedding.embeddings
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocab_size": self.vocab_size,
+                "hidden_size": self.hidden_size,
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "max_sequence_length": self.max_sequence_length,
+                "num_segments": self.num_segments,
+                "intermediate_dim": self.intermediate_dim,
+                "dropout": self.dropout,
+            }
+        )
+        return config
+
+
+class BertClassifier(keras.Model):
+    """
+    Adds a classification head to a Bert encoder model.
```

Comment:
nit: delete verbs - Classification head of Bert model.

Reply:
Move summary line to be the first line in the docstring.

Reply:
Not to nit a nit but this isn't a classification head---it's a full tokens to logits model! I've tried to amend the comment for clarity.